tests/integration/test_calico_integration.py

```diff
@@ -181,3 +181,30 @@ async def test_bgp_service_ip_advertisement(ops_test, bird_charm, kubernetes):
         'global-bgp-peers': '[]'
     })
     await bird_app.destroy()
+
+
+async def test_rp_filter_conflict(ops_test):
+    unit_number = 0
+    retcode, stdout, stderr = await ops_test.run(
+        'juju', 'ssh', '-m', ops_test.model_full_name, f'calico/{unit_number}',
+        'sudo', 'sysctl', '-w', 'net.ipv4.conf.all.rp_filter=2'
+    )
+    if retcode != 0:
+        pytest.fail(f"Failed to set rp_filter\n stdout: {stdout}\n stderr: {stderr}")
```

Comment:
`ops_test` has a `juju` method for this sort of thing.  
`ops_test` automatically applies the correct model for you so there's no need for `-m <model>`.  
The `sudo cmd...` i think should be considered all one argument 
`ops_test.run()` has a `check=False` which does what you're doing below and even lets you set your own `fail_msg`

```suggestion
    await ops_test.juju(
        'ssh',  f'calico/{unit_number}', 'sudo sysctl -w net.ipv4.conf.all.rp_filter=2',
        check=True,
        fail_msg="Failed to set rp_filter"
    )
```

Reply:
newest commit uses this juju function instead